Misc/NEWS.d/next/Library/2024-10-05-11-54-31.gh-issue-124927.OOOlif.rst

```diff
@@ -0,0 +1 @@
+The setpos_from_xy function and pos2xy take into account the zeros in reader.screeninfo and treat characters (such as U+200C, show as "\u200c" on REPL) occupying multiple grids as a single Position.
```

Comment:
The NEWS entry can be more concise. The repl implementation is a details and not exposed publicly (or am I wrong?) so you can just say that non-printing characters are now properly handled.

By the way, what happens with ANSI escape sequences that are by default non-printing? can you test the case?

Reply:
The NEWS has been updated, Thank you.

> By the way, what happens with ANSI escape sequences that are by default non-printing? can you test the case?

https://github.com/python/cpython/blob/16cd6cc86b3ba20074ae3eefb61aeb24ee1544f7/Lib/_pyrepl/reader.py#L42-L69
If I didn't misunderstand your meaning, it seems that all non-printing characters belong to the Cf category, so the disp_str functions can correctly identify and escape them to printable.